src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -389,8 +389,10 @@ private function evaluate( $directive_value, string $default_namespace, $context
 		$path_segments = explode( '.', $path );
 		$current       = $store;
 		foreach ( $path_segments as $path_segment ) {
-			if ( isset( $current[ $path_segment ] ) ) {
+			if ( is_array( $current ) && isset( $current[ $path_segment ] ) ) {
```

Comment:
I'm tempted to include a check like `( is_array( $current ) || $current instanceof ArrayAccess )` here 🤔

It's probably useful only in exceptional cases, but it does seem like the most correct way to check.

Do other folks have thoughts on that?

Reply:
It looks good to me, I will make these changes.

Reply:
Thank you! My biggest concern at this point is that this behavior has no tests. Can you work on some test cases for this?

This is an example (or other files in this directory):

https://github.com/WordPress/wordpress-develop/blob/c7db7886b81da3e600dc5e6c9c961c54d1d643ef/tests/phpunit/tests/interactivity-api/wpInteractivityAPI-wp-text.php#L32-L43

Reply:
I added a test in https://github.com/WordPress/wordpress-develop/pull/6672
It should go in `evaluate()`  function testing.